boot/makebootable.go

```diff
@@ -357,6 +359,18 @@ func copyBootSnap(orig string, dstInfo *snap.Info, dstSnapBlobDir string) error
 	return nil
 }
 
+func cryptsetupSupportsTokenReplaceImpl() bool {
+	cmd := exec.Command("cryptsetup", "--test-args", "token", "import", "--token-id", "0", "--token-replace", "/dev/null")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		logger.Noticef("WARNING: cryptsetup does not support option --token-replace: %v: %s", err, out)
+		return false
+	}
+	return true
+}
```

Comment:
@pedronis I have been trying to find a way to know when to enable writing key data to key slot tokens. I was thinking of using the model, but that does not work. For one classic, even 24.10 uses "base: core22". Not sure why. The "base" also do not make much sense in classic anyway.

So for now, I check that cryptsetup supports `--token-replace`. But maybe we can refine it. For example, if the model is not classic, then we could just say that we disable the feature when base is core20 or core22. But for classic, we just check cryptsetup.

Do you have any opinion about that?

Reply:
in theory classic should also use the base correctly but I think they had issues to migrate some things to core24? but basically I think we should just start supporting:

* classic + token-replace

after we are happy with that we should consider

* UC with core24+